app/src/lib/git/submodule.ts

```diff
@@ -1,10 +1,24 @@
 import { git } from './core'
 import { Repository } from '../../models/repository'
 import { SubmoduleEntry } from '../../models/submodule'
+import { pathExists } from '../file-system'
+import * as Path from 'path'
 
 export async function listSubmodules(
   repository: Repository
 ): Promise<ReadonlyArray<SubmoduleEntry>> {
+  const submodulesFile = await pathExists(
+    Path.join(repository.path, '.gitmodules')
+  )
+  const submodulesDir = await pathExists(
```

Comment:
Could we run these two checks in parallel rather than sequentially please?

```ts
const [ submodulesFile, subModulesDir ] = await Promise.all([
  pathExists(Path.join(repository.path, 'gitmodules'),
  pathExists(Path.join(repository.path, 'git', 'modules'),
])
```

Reply:
Solid recommendation. Thanks! Done!